internal/pkg/term/syncbuffer/syncbuffer.go

```diff
@@ -0,0 +1,99 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package syncbuffer provides a goroutine safe bytes.Buffer as well printing functionality to the terminal.
+package syncbuffer
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+)
+
+// SyncBuffer is a synchronized buffer that can be used to store output data and coordinate between multiple goroutines.
+type SyncBuffer struct {
```

Comment:
i wonder if we need to export `SyncBuffer` _and_ `LabeledSyncBuffer`? Seems like we'll only ever use the `LabeledSyncBuffer` (correct me if i'm wrong!)

Reply:
As of Now for `Build And Push` we only use the `Labeled SyncBuffer`, may be in future for some other use cases we can use the `SyncBuffer` with out any label.

Reply:
i'd say let's not export it until we find out it's needed! not a huge deal either way since it's an internal package, though.